### PR TITLE
Migrate to MUIv5

### LINKED ISSUE #328: Migrate fully to React 18
**Is your feature request related to a problem? Please describe.**
We can't run `npm i` without having to pass the `--legacy-deps` flag. Somewhere, our React dependencies got out of sync, so we should be using React 18 anyway.

**Describe the solution you'd like**
Move to React 18 fully

**Describe alternatives you've considered**
continue passing the lagacy deps flag

**Additional context**
See: https://react.dev/blog/2022/03/08/react-18-upgrade-guide

### PR DESCRIPTION
**Pull request by marvinsjsu**
fixes #328 

## Dev Summary

This PR has changes for migrating to MUIv5.  Manual testing is needed to make sure this hasn't broken anything.

## Test Plan

repro steps:

1. localhost:3000; we may need to:
  -  `rm -rf node_modules package-lock.json`
  -  `npm i`
2. click through the pages and UI elements (buttons, dropdowns, etc.) of the app
and verify  that look and functionality in our localhost matches that of https://npc-fm7ocfzuva-uw.a.run.app

## Resources

- [Migrating to MUIv5](https://mui.com/material-ui/migration/migration-v4/)

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -65,14 +65,14 @@
 		"@types/react-router-dom": "^5.1.7",
 		"@typescript-eslint/eslint-plugin": "^5.5.0",
 		"@typescript-eslint/parser": "^5.5.0",
-		"react-scripts": "^5.0.1",
```
Comment (esteban-gs) [inline client/package.json:68]: We don't need this anymore since we are still using CRA (Create React App)?

File: client/package.json [hunk 2]
```diff
@@ -3,13 +3,13 @@
 	"version": "0.1.0",
 	"private": true,
 	"dependencies": {
-		"@emotion/react": "^11.9.0",
-		"@emotion/styled": "^11.8.1",
+		"@emotion/react": "^11.11.0",
+		"@emotion/styled": "^11.11.0",
 		"@hookform/resolvers": "^2.9.7",
-		"@material-ui/core": "^4.12.4",
+		"@mui/base": "^5.0.0-beta.2",
```
Comment (esteban-gs) [inline client/package.json:9]: It seems like this isn't used anywhere, so probably best to uninstall it.

File: client/src/types/material-ui.d.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ declare module '@mui/material/styles' {
     };
   }
 
-  interface ThemeOptions {
+  interface DeprecatedThemeOptions {
```
Comment (esteban-gs) [inline client/src/types/material-ui.d.ts:10]: @marvinsjsu , does this mean our theme doesn't work anymore?

**Review by esteban-gs — APPROVED** [review]
@marvinsjsu , this looks good to me. We'll have to keep an eye out for any pages that break, but I think we should be ready to merge. I just have a couple of questions.

File: client/package.json (hunk 1 above)
Comment (marvinsjsu) [reply client/package.json:68]: Oh, yea we still need this ... it just got moved to line 75 - https://github.com/Nonprofit-Exchange-Hub/web-app/blob/19f13606991b3f6f2e940960eba41f070fb841a9/client/package.json#L75

File: client/src/types/material-ui.d.ts (hunk 3 above)
Comment (marvinsjsu) [reply client/src/types/material-ui.d.ts:10]: This change looks to be from the MUI5 package, but the themes should still work.  I've tested by updating some values on `client/src/theme/palette.ts` (for example changed text.primary to '#FF0000').

**Pull request merged**